crates/static-analysis-kernel/src/analysis/generated_content.rs

```diff
@@ -0,0 +1,154 @@
+/// Some heuristics are based on these sources
+///  - https://github.com/github-linguist/linguist/blob/master/lib/linguist/generated.rb
+///
+/// We only look at the first few bytes of the code that are generally comments generated by
+/// code generation tools. We look at most at [MAX_HEADER_SIZE] characters.
+pub fn is_generated_file(content: &str, language: &Language) -> bool {
+    match language {
+        Language::Go => {
+            if content.contains("Code generated by MockGen") {
+                return true;
+            }
+            if content.contains("Code generated by") {
+                return true;
+            }
+            if content.contains(PROTOBUF_HEADER) {
+                return true;
+            }
+            if content.contains(THRIFT_HEADER) {
+                return true;
+            }
+            false
+        }
+        Language::Java => {
+            if content.contains("generated by the protocol buffer compiler") {
+                return true;
+            }
+            if content.contains(PROTOBUF_HEADER) {
+                return true;
+            }
+            if content.contains(THRIFT_HEADER) {
+                return true;
+            }
+            false
+        }
+        Language::JavaScript => {
+            if content.contains("Generated by PEG.js") {
+                return true;
+            }
+            if content.contains("GENERATED CODE -- DO NOT EDIT!") {
+                return true;
+            }
+            if content.contains(THRIFT_HEADER) {
+                return true;
+            }
+            false
+        }
+        Language::Python => {
+            if content.contains("Generated protocol buffer code") {
+                return true;
+            }
+            if content.contains("Code generated by") {
+                return true;
+            }
+            if content.contains(PROTOBUF_HEADER) {
+                return true;
+            }
+            if content.contains(THRIFT_HEADER) {
+                return true;
+            }
+            false
+        }
+        Language::Ruby => {
+            if content.contains(PROTOBUF_HEADER) {
+                return true;
+            }
+            if content.contains(THRIFT_HEADER) {
+                return true;
+            }
+            false
+        }
+        Language::TypeScript => {
+            if content.contains("Generated by PEG.js") {
+                return true;
+            }
+            if content.contains("GENERATED CODE -- DO NOT EDIT!") {
+                return true;
+            }
+            if content.contains(THRIFT_HEADER) {
+                return true;
+            }
+            false
+        }
+        _ => false,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::analysis::generated_content::is_generated_file;
+    use crate::model::common::Language;
+
+    #[test]
+    fn test_is_generated_file_java() {
```

Comment:
These tests are here because a lot of bugs are due to small syntactical changes that are not caught during review and break functionalities. I did not want to have files because it would add more files to the repo only for tests. Those tests do not handle all cases but they will catch potential changes in the most used languages and I think we should keep them for this reason (e.g. prevent feature breakage).